crates/viewer/re_view_spatial/tests/mesh.rs

```diff
@@ -0,0 +1,58 @@
+use re_log_types::TimePoint;
+use re_test_context::{TestContext, external::egui_kittest::SnapshotOptions};
+use re_test_viewport::TestContextExt as _;
+use re_types::{RowId, archetypes::Mesh3D};
+use re_viewer_context::{RecommendedView, ViewClass as _};
+use re_viewport_blueprint::ViewBlueprint;
+
+#[test]
+pub fn test_single_channel_mesh() {
+    let texture_format = re_types::components::ImageFormat::l8([2, 2]);
+    let texture_buffer = re_types::components::ImageBuffer::from(vec![128, 255, 0, 128]);
+    let mut test_context = TestContext::new_with_view_class::<re_view_spatial::SpatialView3D>();
+
+    // Log a simple quad mesh with a texture with one channel.
+    test_context.log_entity("world/mesh", |builder| {
+        builder.with_archetype(
+            RowId::new(),
+            TimePoint::default(),
+            &Mesh3D::new([
+                [-1.0, 0.0, -1.0],
+                [1.0, 0.0, -1.0],
+                [-1.0, 0.0, 1.0],
+                [1.0, 0.0, 1.0],
+            ])
+            .with_vertex_texcoords([[0.0, 0.0], [1.0, 0.0], [0.0, 1.0], [1.0, 1.0]])
+            .with_triangle_indices([[0, 1, 2], [2, 1, 3]])
+            .with_albedo_texture(texture_format, texture_buffer),
+        )
+    });
+
+    let view_id = test_context.setup_viewport_blueprint(|_ctx, blueprint| {
+        let view_blueprint = ViewBlueprint::new(
+            re_view_spatial::SpatialView3D::identifier(),
+            RecommendedView::root(),
+        );
+
+        let view_id = view_blueprint.id;
+
+        blueprint.add_views(std::iter::once(view_blueprint), None, None);
+
+        view_id
+    });
+
+    let size = egui::vec2(300.0, 300.0);
+
+    let mut harness = test_context
+        .setup_kittest_for_rendering()
+        .with_size(size)
+        .build_ui(|ui| test_context.run_with_single_view(ui, view_id));
+
+    let broken_pixels_fraction = 0.0045;
```

Comment:
c&p from another test or is this empircal result from running ci?
we shouldn't use this whenever possible, there's already some default threshholding going on which tries to be edge sensitive, so chances we could run this without extra threshhold

(naturally the higher the threshhold the less often we catch buts, the lower it is the more flakiness we risk)

Reply:
those error rates are super mysterious unfortunately. I haven't really gotten my head around yet what they actually mean >.<
it's not a mean-squared-error as most image comparisons would do, instead it's using https://crates.io/crates/dify to mark pixel as either "success" or "broken". The main threshhold controls the "brokenness" (???) per pixel in dify and then we added "completely broken pixels" on top because ofc some pixels are just completely off.
Dify seems to do a great job being tolerant on edges which I like, but otherwise the controls are unfortunately super intransparent right now :/